File: d1/d1.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

int main() {
    freopen("d1.in", "r", stdin);
    string line;
    map<string, string> rep = {
        {"one", "o1e"},
        {"two", "t2o"},
        {"three", "t3e"},
        {"four", "f4r"},
        {"five", "f5e"},
        {"six", "s6x"},
        {"seven", "s7n"},
        {"eight", "e8t"},
        {"nine", "n9e"},
        {"ten", "z0o"}
    };
    int p1 = 0, p2 = 0;
    while (getline(cin, line)) {
        string s1 = "";
        for (char c : line) {
            if (c-'0' >= 0 && c-'0' <= 9) {
                s1 += c;
            }
        }
        string s2 = "";
        for (auto& [from, to] : rep) {
            line = regex_replace(line, regex(from), to);
        }
        for (char c : line) {
            if (c-'0' >= 0 && c-'0' <= 9) {
                s2 += c;
            }
        }
        p1 += 10*(s1[0]-'0') + s1[s1.size()-1]-'0';
        p2 += 10*(s2[0]-'0') + s2[s2.size()-1]-'0';
    }
    cout << p1 << ' ' << p2 << '\n';
}

File: d1/d1.py
import sys
import re
import itertools
import collections
from collections import defaultdict
from collections import Counter

io = sys.argv[1] if len(sys.argv) > 1 else "d1.in"

inp = [x for x in open(io).read().strip().splitlines()]
#print(inp)

nums_conv = {
    "one": 1, "two": 2, "three": 3, "four": 4, "five" : 5, "six": 6, "seven": 7, "eight": 8, "nine": 9, "zero": 0,
    "1": 1, "2": 2, "3": 3, "4": 4, "5" : 5, "6": 6, "7": 7, "8": 8, "9": 9, "0": 0
}

sum = 0
for line in inp:
    s = list(filter(lambda x: x.isdigit(), line))
    sum += int(s[0] + s[-1])
print(sum)

sum = 0
for line in inp:
    s = []
    for i in range(len(line)):
        for n in nums_conv:
            if line[i:].startswith(n):
                s.append(nums_conv[n])
    sum += 10*s[0] + s[-1]

print(sum)
